src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -174,30 +127,19 @@ public bool TryGetValue(object key, out object value)
                 if (_entries.TryGetValue(key, out entry))
                 {
                     // Check if expired due to expiration tokens, timers, etc. and if so, remove it.
-                    if (entry.CheckExpired(_clock.UtcNow))
+                    if (entry.CheckExpired(utcNow))
                     {
                         expiredEntry = entry;
                     }
                     else
                     {
-                        // Refresh sliding expiration, etc.
-                        entry.LastAccessed = _clock.UtcNow;
-                        value = entry.Value;
                         found = true;
+                        entry.LastAccessed = utcNow;
+                        result = entry.Value;
 
-                        var link = EntryLinkHelpers.ContextLink;
-                        if (link != null)
-                        {
-                            // Copy expiration tokens and AbsoluteExpiration to the link
-                            if (entry.Options.ExpirationTokens != null)
-                            {
-                                link.AddExpirationTokens(entry.Options.ExpirationTokens);
-                            }
-                            if (entry.Options.AbsoluteExpiration.HasValue)
-                            {
-                                link.SetAbsoluteExpiration(entry.Options.AbsoluteExpiration.Value);
-                            }
-                        }
+                        // When an entry is returned from an ambient entry,
```

Comment:
This comment doesn't make sense.
When this entry is retrieved in the scope of creating another entry, that entry needs a copy of these expiration tokens.